std/algorithm/searching.d

```diff
@@ -1784,35 +1788,54 @@ if (isRandomAccessRange!R1 && isBidirectionalRange!R2
     // auto needleBack = moveBack(needle);
     // Stage 1: find the step
     size_t step = 1;
-    auto needleBack = needle.back;
+    immutable needleBack = needle.back;
     needle.popBack();
     for (auto i = needle.save; !i.empty && i.back != needleBack;
          i.popBack(), ++step)
     {
     }
     // Stage 2: linear find
     size_t scout = needleLength - 1;
+outer:
     for (;;)
     {
         if (scout >= haystack.length)
         {
-            return haystack[haystack.length .. haystack.length];
+            return haystack[$ .. $];
```

Comment:
> The code already uses haystack[$ .. $]. This is a latent bug then?

Yeah, i'd say so.

> Worth a more thorough inspection, issue, and separate PR?

For the rest of Phobos, yeah, but you can go ahead and fix it in this PR. I honestly have never thought about it before this PR, but I'd bet there are tons of uses of `$` like this in Phobos that need to be fixed.